salt/metalk8s/kubernetes/coredns/deployed.sls

```diff
@@ -17,15 +17,15 @@ Create coredns ConfigMap:
                 health
                 kubernetes {{ coredns.cluster_domain }} {{ coredns.reverse_cidrs }} {
                   pods insecure
-                  upstream
                   fallthrough in-addr.arpa ip6.arpa
                 }
                 prometheus :9153
-                proxy . /etc/resolv.conf
+                forward . /etc/resolv.conf
                 cache 30
                 loop
                 reload
                 loadbalance
+                ready
             }
```

Comment:
It appears these changes are made using some tool? However, they're not (necessarily) in line with what `kubeadm` now deploys: https://github.com/kubernetes/kubernetes/blob/87473a5d27e52f8eb86575898cd6e6d8b8e881be/cmd/kubeadm/app/phases/addons/dns/manifests.go#L303 (note: this is `master`, should figure out what `kubeadm` does for the K8s release we want to ship).

Would make sense to re-run/update https://github.com/NicolasT/kubeadm-dry-run

Reply:
Ideally here,  it's a 2.5 branch and we ship K8S 1.16.8.

Cross-checking with the `kubeadm` specs from a corresponding release here: https://github.com/kubernetes/kubernetes/blob/release-1.16/cmd/kubeadm/app/phases/addons/dns/manifests.go#L306, It seems we are all good. 

Note: One thing which the `kubeadm` Corefile adds is a TTL set to 30 seconds. I will add this and be done.

Reply:
I would use the same content as what `kubeadm` deploys. Indeed, `ready` is in there, but at a different location in the file. Would be easier to 'review' if we just deploy the exact same thing.

Furthermore, the RBAC resources may need to be validated as well?

Reply:
Yeah, I followed same placement as that of `kubeadm`

Reply:
+ all good with RBACn definitions

Reply:
> Would make sense to re-run/update https://github.com/NicolasT/kubeadm-dry-run

Definitely, even make it part of our procedure for updating K8s dependencies. Could we somehow use this as a tool for rendering our formulas, just as we do for Helm?
Also, in the future I think it'd make sense to use the [kubeadm Operator](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/kubeadm/20190916-kubeadm-operator.md) (once it is implemented) directly in the product (would make it much easier to maintain).

Reply:
Not convinced about the latter. Adding another tool which, next to Salt, does things on hosts will result in a messy situation where we no longer know what does what.